### PR TITLE
Functional tests for Admin BlockAccess AB#15412

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#15412](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15412)

## Description

1. Added functional tests for Admin block access feature.

## Testing

- [ ] Unit Tests Updated
- [X] Functional Tests Updated
- [ ] Not Required

## Notes
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js [hunk 1]
```diff
@@ -2,6 +2,32 @@ import { performSearch } from "../../utilities/supportUtilities";
...
+            .then((text) => {
+                const regex = /\((\d+)\)/g;
+                const match = regex.exec(text);
+                historyCount = Number(match[1]);
+                if (expectedRows != null && expectedRows >= 0) {
+                    expect(historyCount).to.equal(expectedRows);
+                }
+                resolve(historyCount);
+            });
+    });
```
Comment (BrianMaki) [inline Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:29]: I am not sure about this type of complexity.  @MikeLyttle  what do you think?

**Review by BrianMaki — APPROVED** [review]
See my one one comment.

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js (hunk 1 above)
Comment (BronzBierd) [reply Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:29]: great this is a good suggestion. I'll simplify

Comment (BronzBierd) [reply Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js]: I've simplified the approach, as its in a table it was easiest just to put the count in the span.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js [hunk 2]
```diff
@@ -2,6 +2,20 @@ import { performSearch } from "../../utilities/supportUtilities";
...
+
+function checkAgentAuditHistory() {
+    cy.get("[data-testid=agent-audit-history-title]")
+        .should("be.visible")
+        .click();
+
+    cy.get("[data-testid=agent-audit-history-table]").should("be.visible");
+
+    let historyCount = 0;
+    return cy.get("[data-testid=agent-audit-history-count]").invoke("text");
```
[the remark this answers is not in the document]
Comment (BronzBierd) [reply Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:17]: Thanks for catching this oversight.

**Pull request merged**